File: src/main/java/com/example/ajaxonboardingproject/service/mapper/MovieMapper.java
package com.example.ajaxonboardingproject.service.mapper;

import com.example.ajaxonboardingproject.dto.request.MovieRequestDto;
import com.example.ajaxonboardingproject.dto.response.MovieResponseDto;
import com.example.ajaxonboardingproject.model.Movie;
import org.springframework.stereotype.Component;

@Component
public class MovieMapper implements RequestDtoMapper<MovieRequestDto, Movie>,
        ResponseDtoMapper<MovieResponseDto, Movie> {
    @Override
    public Movie mapToModel(MovieRequestDto dto) {
        Movie movie = new Movie();
        movie.setTitle(dto.getTitle());
        movie.setDescription(dto.getDescription());
        return movie;
    }

    @Override
    public MovieResponseDto mapToDto(Movie movie) {
        return new MovieResponseDto(
        movie.getId(),
        movie.getTitle(),
        movie.getDescription());
    }
}

File: src/main/java/com/example/ajaxonboardingproject/service/mapper/MovieMapper.kt
package com.example.ajaxonboardingproject.service.mapper

import com.example.ajaxonboardingproject.dto.request.MovieRequestDto
import com.example.ajaxonboardingproject.dto.response.MovieResponseDto
import com.example.ajaxonboardingproject.model.Movie
import org.springframework.stereotype.Component

@Component
class MovieMapper : RequestDtoMapper<MovieRequestDto, Movie>,
    ResponseDtoMapper<MovieResponseDto, Movie> {
    override fun mapToModel(dto: MovieRequestDto): Movie {
        return Movie(
            title = dto.title,
            description = dto.description
        )
    }

    override fun mapToDto(model: Movie): MovieResponseDto {
        return MovieResponseDto(
            id = model.id,
            title = model.title,
            description = model.description
        )
    }
}
